### ISSUE TITLE
Tensorboard TypeError when using flag with null and numbers

### ISSUE THREAD

**Issue by sxooler** [guildai/guildai#163]
## Current setup
**guildai**: 0.7.0rc9
**Python**: 3.8.2
**OS**: Win 10

## Issue

I have run this command (I'm using custom Guild home path as I've run into issues when running Guild from my D drive and having Guild home on C):

```
guild -H .guild tensorboard
```

TypeError traceback: https://pastebin.com/PyHwMbMu

## Comments

I found that when I remove some specific runs from my Guild home, tensorboard runs without any issues. So I dive into comparison of these particular runs to look for the issue - what do they have in common?

As I'm using global Python vars, my `train.py` file (which is my main file used by guild) looks like this:

```python
import mylib
# and other imports

L1=5
L2=None
# and other global vars

forecaster = mylib.forecaster(L1, L2, ...)
forecaster. run()
forecaster.eval()
# and few other commands
```

Everything works fine. But when I run e.g. these two trainings (I have single operation so I do not have to specify it):
```
guild -H .guild run L1=10
guild -H .guild run L1=10 L2=5
```
Then I have an issue with tensorboard. It seems, it has issue with type mixing (L2 can be *None* or *integer*). Any fix would be appreciated.

**Comment by gar1t** [comment]
Thanks for the report and sorry for the long delay in responding!

I creates a project to try to recreate this:

https://github.com/guildai/issue-resolution/tree/master/163-tensorboard-typeerror-when-using-flag-with-null-and-numbers

 So far I can't reproduce the error. I suspect we'll need to dig into how L1 and L2 are being used.

The impact has to be through scalars as that's where the error is. Something is logging something that's causing this crash. Posting an event file would help, but ideally we would update that project with some steps to reproduce.

This strikes me as similar to #99.

**Comment by sxooler** [comment]
You can easily reproduce the error with the created project:
```
guild run test.py
guild run test.py L2=5
guild tensorboard
```
The first will set L2 to default (which is None) and the second run set it to value 5 (integer). Everything works fine but when you run tensorboard, you get the error I pasted in my question.

I think you're right that it's very similar to mentioned issue, my feeling is there might be some issue with type mixing. The error says `TypeError: dtype mismatch: not isinstance('', int)`, maybe `None` value stored (or load) as an empty string which causes `TypeError`?

**Comment by gar1t** [comment]
I confirmed this is fixed in 0.7. Refer to the issue doc above for the steps I used to recreate. Thanks again for the report!
